### PR TITLE
Deprecate trace mutli propagator

### LINKED ISSUE #2244: TraceMultiPropagator vs. MultiTextMapPropagator
There are two very similar ways and classes that allow multiple trace context propagators to be in play at the same time:
* TraceMultiPropagator - lives in extensions module, created by invoking `TraceMultiPropagator.create(...)`
* MultiTextMapPropagator - created by invoking `TextMapPropagator.composite(...)`

The implementations are very similar -- they both allow context injection and extraction by delegating to a list of other propagators.  However, there is a subtle difference between them,  due to the fact that `TraceMultiPropagator.extract()` stops extracting after the first success.

This subtle difference could be confusing to users.  
Since it does seem likely that users will want/need to support multiple formats, we should keep this...but we should avoid the duplicate implementation.  

Let's remove the `TraceMultiPropagator` and add a "stopAfterFirstFound" flag to the `MultiTextMapPropagator` constructor that will be referenced when extracting.  This should default to `false` to preserve the existing behavior.  
The implementation should also beef up the javadoc in `MultiTextMapPropagator`.

### PR DESCRIPTION
**Pull request by breedx-splk**
This resolves #2244...even if the implementation here doesn't perfectly match what's in that issue.  There were discussions on the previous PR (#2273) and we agreed to deprecate the extension class and remove it in the next release.  Since 0.13.0 went out today, I assumed we could remove it in 0.14.0 but maybe people think 0.15.0 is better?

In any case, I also brought some missing tests over, hopefully coverage goes up slightly.

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java [hunk 1]
```diff
@@ -48,8 +48,12 @@
  * Context context = OpenTelemetry.getPropagators().getTextMapPropagator()
  *   .extract(context, carrier, carrierGetter);
  * }</pre>
+ *
+ * This class is deprecated and will be removed in 0.14.0. Users should migrate to the
```
Comment (anuraaga) [inline extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java:52]: Oops nit - `@deprecated -`

Comment (breedx-splk) [reply extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java]: I guess that makes it easier to find?  Anyway, added that `@`...

File: extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java [hunk 2]
```diff
@@ -49,7 +49,7 @@
  *   .extract(context, carrier, carrierGetter);
  * }</pre>
  *
- * This class is deprecated and will be removed in 0.14.0. Users should migrate to the
+ * This class is @deprecated and will be removed in 0.14.0. Users should migrate to the
```
Comment (jkwatson) [inline extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java:52]: I think @anuraaga was asking to add a `@deprecated` tag to the javadoc itself, not embed the `@` in the comment.

File: extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java [hunk 3]
```diff
@@ -49,7 +49,7 @@
...
  * MultiTextMapPropagator in the api.
  */
```
Comment (jkwatson) [inline extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java:54]:
```suggestion
 * @deprecated Migrate to the MultiTextPropagator in the API.
 */
```

Comment (anuraaga) [reply extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java:54]: Yeah sorry for lack of clarity. This format will render a bit better in IntelliJ I think

**Review by jkwatson — APPROVED** [review]
Thanks!

**Comment by jkwatson** [comment]
looks like something is grumbling about using deprecated methods in the build. Probably tests, I'd reckon.

File: api/context/src/test/java/io/opentelemetry/context/propagation/MultiTextMapPropagatorTest.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class MultiTextMapPropagatorTest {
```
Comment (anuraaga) [inline api/context/src/test/java/io/opentelemetry/context/propagation/MultiTextMapPropagatorTest.java:27]:
```suggestion
@SuppressWarnings("deprecation")
class MultiTextMapPropagatorTest {
```

File: extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java [hunk 5]
```diff
@@ -48,8 +48,13 @@
  * Context context = OpenTelemetry.getPropagators().getTextMapPropagator()
  *   .extract(context, carrier, carrierGetter);
  * }</pre>
+ *
+ * This class is @deprecated and will be removed in 0.14.0. Users should migrate to the
+ * MultiTextMapPropagator in the api.
+ * @deprecated Migrate to the MultiTextPropagator in the API.
```
Comment (anuraaga) [inline extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/TraceMultiPropagator.java:54]:
```suggestion
 * @deprecated Use {@link TextMapPropagator#composite}
```

**Review by anuraaga — APPROVED** [review]

File: api/context/src/test/java/io/opentelemetry/context/propagation/MultiTextMapPropagatorTest.java [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@SuppressWarnings("deprecation")
```
Comment (jkwatson) [inline api/context/src/test/java/io/opentelemetry/context/propagation/MultiTextMapPropagatorTest.java:27]:
```suggestion
@SuppressWarnings("deprecation") // we're testing deprecated stuff in here.
```

**Comment by jkwatson** [comment]
looks like spotless needs a little more help on one of the tests

**Pull request merged**